tests/run.sh

```diff
@@ -53,7 +61,13 @@ if [[ ! -f broken_update.ext4 ]]; then
     dd if=/dev/urandom of=broken_update.ext4 bs=10M count=5
 fi
 
+if [[ $1 = "slow" ]]; then
+    run_slow_tests
+fi
 
+if [[ $1 = "fast" ]]; then
+    run_fast_tests
+fi
 
-py.test --maxfail=1 -s --tb=short --runfast --gateway "${GATEWAY_IP_PORT}" --clients "${CLIENT_IP_PORT}" --verbose --junitxml=results.xml tests/{test_bootstrapping.py,test_basic_integration.py,test_image_update_failures.py,test_fault_tolerance.py}
-py.test --maxfail=1 -s --tb=short --runslow --gateway "${GATEWAY_IP_PORT}" --clients "${CLIENT_IP_PORT}" --verbose --junitxml=results.xml tests/{test_basic_integration.py,test_image_update_failures.py,test_fault_tolerance.py}
+run_slow_tests
+run_fast_tests
```

Comment:
You should rearrange all of these into 'if/elif/else'. Right now you'll run one of them, and then still run both again afterwards.